### ISSUE TITLE
istioctl commands failing in 1.7.0 and 1.7.1

### ISSUE THREAD

**Issue by andrewridout** [istio/istio#27383]
**Bug description**
Using commands like `istioctl version` or `istioctl ps` fail when using the Istio 1.7.0 and 1.7.1 `istioctl` client:
```
$ istioctl version
1 error occurred:
        * error port-forewarding into istiod-6c66b6c744-5hdvh : the server is currently unable to handle the request (get pods istiod-6c66b6c744-5hdvh:15014)


1.7.1
```
```
$ istioctl ps
Error: the server is currently unable to handle the request (get pods istiod-6c66b6c744-5hdvh:8080)
```

However, if I use the v1.6.9 version of istioctl client, the commands work:
```
$ istio-1.6.9/bin/istioctl version
client version: 1.6.9
control plane version: 1.7.1
data plane version: 1.7.1 (9 proxies), 1.6.9 (1 proxies)
```
```
$ istio-1.6.9/bin/istioctl ps
NAME                                                   CDS        LDS        EDS        RDS        PILOT                       VERSION
istio-ingressgateway-6f59657d45-sn4fc.istio-system     SYNCED     SYNCED     SYNCED     SYNCED     istiod-6c66b6c744-5hdvh     1.7.1
prometheus-59bb87b99d-bs25x.istio-system               SYNCED     SYNCED     SYNCED     SYNCED     istiod-6c66b6c744-5hdvh     1.6.9
```

[ ] Docs
[x] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
For all `istioctl` commands to work when using versions 1.7.0 and 1.7.1.

**Steps to reproduce the bug**
1. Download `istio-1.6.9` and run `istioctl version`. This will work and output the version.
2. Download `istio-1.7.0` or `istio.1.7.1` and run `istioctl version`. This will fail with the above error message.

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
- Istio version 1.7.0 and 1.7.1
```
$ kubectl version --short
Client Version: v1.18.3
Server Version: v1.16.13-gke.401
```

**How was Istio installed?**
Originally installed with `istioctl manifest apply -f istio-config.yaml`.
Upgraded with `istioctl upgrade -f istio-config.yaml`

**Environment where bug was observed (cloud vendor, OS, etc)**
Google Kubernetes Engine (private cluster) 1.16.13-gke.401

**Comment by howardjohn** [comment]
Can you follow the steps here? https://github.com/istio/istio/issues/27359#issuecomment-693727041

**Comment by andrewridout** [comment]
@howardjohn Here's the output:
```
$ kubectl get --raw '/api/v1/namespaces/istio-system/pods/istiod-6c66b6c744-5hdvh:15014/proxy/debug' -v9
I0917 16:38:24.946521   78662 loader.go:375] Config loaded from file:  /Users/andrew/.kube/config
I0917 16:38:24.949358   78662 round_trippers.go:423] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.18.3 (darwin/amd64) kubernetes/2e7996e" 'https://35.205.19.226/api/v1/namespaces/istio-system/pods/istiod-6c66b6c744-5hdvh:15014/proxy/debug'
I0917 16:38:55.119677   78662 round_trippers.go:443] GET https://35.205.19.226/api/v1/namespaces/istio-system/pods/istiod-6c66b6c744-5hdvh:15014/proxy/debug 503 Service Unavailable in 30170 milliseconds
I0917 16:38:55.119699   78662 round_trippers.go:449] Response Headers:
I0917 16:38:55.119703   78662 round_trippers.go:452]     Audit-Id: 83c2a097-2bc5-4b43-ac4f-86e8f1bc0d2d
I0917 16:38:55.119706   78662 round_trippers.go:452]     Cache-Control: no-cache, private
I0917 16:38:55.119709   78662 round_trippers.go:452]     Content-Type: text/plain; charset=utf-8
I0917 16:38:55.119711   78662 round_trippers.go:452]     X-Content-Type-Options: nosniff
I0917 16:38:55.119714   78662 round_trippers.go:452]     Content-Length: 69
I0917 16:38:55.119718   78662 round_trippers.go:452]     Date: Thu, 17 Sep 2020 15:38:55 GMT
I0917 16:38:55.119752   78662 request.go:1068] Response Body: Error trying to reach service: 'dial tcp 10.4.3.6:15014: i/o timeout'
I0917 16:38:55.129662   78662 helpers.go:216] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "the server is currently unable to handle the request",
  "reason": "ServiceUnavailable",
  "details": {
    "causes": [
      {
        "reason": "UnexpectedServerResponse",
        "message": "Error trying to reach service: 'dial tcp 10.4.3.6:15014: i/o timeout'"
      }
    ]
  },
  "code": 503
}]
F0917 16:38:55.131891   78662 helpers.go:115] Error from server (ServiceUnavailable): the server is currently unable to handle the request
```

**Comment by esnible** [comment]
@howardjohn 
- How can I reproduce this?  I want to at least test it manually before writing the fix.
- There is a spelling error: "error port-forewarding ...." in Istio 1.7.x, but not master.

**Comment by howardjohn** [comment]
@esnible I don't know all the different cases but for sure one is if you do not have connectivity from api-server -> istiod:15014. In GKE this can be tuned with firewall rules, not sure how to do so in kind or other setups though

**Comment by upodroid** [comment]
I can confirm the same error.

GKE Cluster


In my case, it was trying to connect over port 8080.

```
 root  DESKTOP-7JUP8RO  mnt  c  …  Git  ccoe-landing-zone  k8s   master  119✎  1+  SIGINT  #    istioctl proxy-status 
Error: the server is currently unable to handle the request (get pods istiod-f58775d55-9gz6g:8080)
 root  DESKTOP-7JUP8RO  mnt  c  …  Git  ccoe-landing-zone  k8s   master  119✎  1+  ERROR  #    kubectl get --raw '/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug' -v9
I0917 22:34:18.912887   21815 loader.go:375] Config loaded from file:  /root/.kube/config
I0917 22:34:18.913598   21815 round_trippers.go:423] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: kubectl/v1.18.8 (linux/amd64) kubernetes/9f2892a" 'https://34.90.202.183/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug'
I0917 22:34:19.228691   21815 round_trippers.go:443] GET https://34.90.202.183/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug 200 OK in 315 milliseconds
I0917 22:34:19.228760   21815 round_trippers.go:449] Response Headers:
I0917 22:34:19.228770   21815 round_trippers.go:452]     Audit-Id: c94cd724-75f0-4045-99fe-7c455468e7f6
I0917 22:34:19.228779   21815 round_trippers.go:452]     Cache-Control: no-cache, private
I0917 22:34:19.228784   21815 round_trippers.go:452]     Content-Type: text/html; charset=utf-8
I0917 22:34:19.228789   21815 round_trippers.go:452]     Date: Thu, 17 Sep 2020 21:34:17 GMT
<html>
<head>
<title>Pilot Debug Console</title>
</head>
<style>
#endpoints {
  font-family: "Trebuchet MS", Arial, Helvetica, sans-serif;
  border-collapse: collapse;
}

#endpoints td, #endpoints th {
  border: 1px solid #ddd;
  padding: 8px;
}

#endpoints tr:nth-child(even){background-color: #f2f2f2;}

#endpoints tr:hover {background-color: #ddd;}

#endpoints th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
  background-color: black;
  color: white;
}
</style>
<body>
<br/>
<table id="endpoints">
<tr><th>Endpoint</th><th>Description</th></tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/adsz">/debug/adsz</a></td><td>Status and debug interface for ADS</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/adsz?push=true">/debug/adsz?push=true</a></td><td>Initiates push of the current state to all connected endpoints</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/authorizationz">/debug/authorizationz</a></td><td>Internal authorization policies</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/cdsz">/debug/cdsz</a></td><td>Status and debug interface for CDS</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/config_distribution">/debug/config_distribution</a></td><td>Version status of all Envoys connected to this Pilot instance</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/config_dump">/debug/config_dump</a></td><td>ConfigDump in the form of the Envoy admin config dump API for passed in proxyID</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/configz">/debug/configz</a></td><td>Debug support for config</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/edsz">/debug/edsz</a></td><td>Status and debug interface for EDS</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/endpointShardz">/debug/endpointShardz</a></td><td>Info about the endpoint shards</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/endpointz">/debug/endpointz</a></td><td>Debug support for endpoints</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/inject">/debug/inject</a></td><td>Active inject template</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/instancesz">/debug/instancesz</a></td><td>Debug support for service instances</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/pprof/">/debug/pprof/</a></td><td>Displays pprof index</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/pprof/cmdline">/debug/pprof/cmdline</a></td><td>The command line invocation of the current program</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/pprof/profile">/debug/pprof/profile</a></td><td>CPU profile</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/pprof/symbol">/debug/pprof/symbol</a></td><td>Symbol looks up the program counters listed in the request</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/pprof/trace">/debug/pprof/trace</a></td><td>A trace of execution of the current program.</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/push_status">/debug/push_status</a></td><td>Last PushContext Details</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/registryz">/debug/registryz</a></td><td>Debug support for registry</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/resourcesz">/debug/resourcesz</a></td><td>Debug support for watched resources</td>
        </tr>

        <tr>
        <td><a href="/api/v1/namespaces/istio-system/pods/istiod-f58775d55-9gz6g:8080/proxy/debug/syncz">/debug/syncz</a></td><td>Synchronization status of all Envoys connected to this Pilot instance</td>
        </tr>

</table>
<br/>
</body>
</html>
 root  DESKTOP-7JUP8RO  mnt  c  …  Git  ccoe-landing-zone  k8s   master  119✎  1+  #    istioctl proxy-status
NAME                                                                        CDS        LDS        EDS        RDS        ISTIOD                     VERSION
activator-68cbc9b5c7-tnttg.knative-serving                                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
alertmanager-prometheus-operator-kube-p-alertmanager-0.default              SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
autoscaler-5cf649dbb-g8lct.knative-serving                                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
cert-manager-79457cf5c5-8rkhs.cert-manager                                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
cert-manager-cainjector-5f845bf6c7-jn6z7.cert-manager                       SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
cert-manager-webhook-8484675bd8-hqsjp.cert-manager                          SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
cluster-local-gateway-85568bc886-n9xsd.istio-system                         SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
controller-56b9b48c6c-vjshs.cloud-run-events                                SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
controller-bc8d75cbc-jncxh.knative-serving                                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
cre-chan-cre-sub-3541cf98-75ad-fe3898e34f97e5dc3a3d8b18bb8vfpdd.default     SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
cre-demo-chan-publish-mk98h-deployment-66cf698d95-h8jss.default             SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
default-brokercell-ingress-6c556fb-4r2c6.cloud-run-events                   SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
event-display-68cdb689b5-884hc.default                                      SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
event-display-7lkvf-deployment-7d55f6565d-rz8gd.default                     SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
eventing-controller-7558c947d-7vbq9.knative-eventing                        SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
eventing-webhook-847f56845b-cw95g.knative-eventing                          SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
istio-ingressgateway-6f59657d45-7mbj2.istio-system                          SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
istio-webhook-6dd89ff45d-bn65g.knative-serving                              SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
net-certmanager-webhook-74f75674c4-zrrfl.knative-serving                    SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
networking-certmanager-79dbf4b7b7-bl8n2.knative-serving                     SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
pingsource-mt-adapter-84b48b94b-vnxpv.knative-eventing                      SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
prometheus-operator-grafana-5d9549f66b-kmndv.default                        SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
prometheus-operator-kube-p-operator-b7cb676cd-6qt76.default                 SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
prometheus-operator-kube-state-metrics-66b4c95cd9-l4jcx.default             SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
prometheus-operator-prometheus-node-exporter-5t54l.default                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
prometheus-operator-prometheus-node-exporter-bh6kp.default                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
prometheus-operator-prometheus-node-exporter-tvpgg.default                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
prometheus-prometheus-operator-kube-p-prometheus-0.default                  SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
thanos-bucket-7cb5b796c7-ndz9s.default                                      SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
thanos-compact-5b584ddc57-tp6bg.default                                     SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
thanos-query-7c454966b5-kbrhv.default                                       SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
thanos-store-0-78ff96ff95-k8d4j.default                                     SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
webhook-85758f4589-b76xx.knative-serving                                    SYNCED     SYNCED     SYNCED     SYNCED     istiod-f58775d55-9gz6g     1.7.1
 root  DESKTOP-7JUP8RO  mnt  c  …  Git  ccoe-landing-zone  k8s   master  119✎  1+  #    istioctl version
1 error occurred:
        * error port-forewarding into cluster-local-gateway-85568bc886-n9xsd : the server is currently unable to handle the request (get pods cluster-local-gateway-85568bc886-n9xsd:15014)


1.7.1
data plane version: 1.7.1 (33 proxies)
 root  DESKTOP-7JUP8RO  mnt  c  …  Git  asml-ccoe-landing-zone  k8s   master  119✎  1+  #    
```

**Comment by upodroid** [comment]
Adding those ports is not documented in https://istio.io/latest/docs/setup/platform-setup/gke/

**Comment by andrewridout** [comment]
@esnible Any idea when this is getting released? I get the same issue with 1.7.3.

**Comment by andrewridout** [comment]
Confirmed that this is now working in Istio 1.7.4.
